### PR TITLE
Nightly build improvements

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:
1. add nightly CDI to HCO nightly build (in addtion to the already exists nightly KubeVirt).

2. Move to FBC

3. Also create and publish nightly index-image

4. Try to deploy the bundle image on a K8s cluster, before publishing the images.

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-48505
```

**Release note**:

```release-note
None
```

### CODE REVIEW

File: automation/nightly/test-nightly-build.sh [hunk 1]
```diff
@@ -57,9 +68,53 @@ HCO_WEBHOOK_IMAGE_DIGEST=$(tools/digester/digester --image ${CSV_WEBHOOK_IMAGE}:
 # Build the CSV
 HCO_OPERATOR_IMAGE=${HCO_OPERATOR_IMAGE_DIGEST} HCO_WEBHOOK_IMAGE=${HCO_WEBHOOK_IMAGE_DIGEST} ./hack/build-manifests.sh
 
-REGISTRY_NAMESPACE=${IMAGE_PREFIX} CONTAINER_TAG=${IMAGE_TAG} make bundleRegistry
+# Download OPM
+OPM_VERSION=v1.47.0
+wget https://github.com/operator-framework/operator-registry/releases/download/${OPM_VERSION}/linux-amd64-opm -O oop
```
Comment (orenc1) [inline automation/nightly/test-nightly-build.sh:73]: do you mean:
```suggestion
wget https://github.com/operator-framework/operator-registry/releases/download/${OPM_VERSION}/linux-amd64-opm -O opm
```
?

File: automation/nightly/test-nightly-build.sh [hunk 2]
```diff
@@ -57,9 +68,53 @@ HCO_WEBHOOK_IMAGE_DIGEST=$(tools/digester/digester --image ${CSV_WEBHOOK_IMAGE}:
...
+
+# deploy the HyperConverged CR
+$KUBECTL apply -n kubevirt-hyperconverged deploy/hco.cr.yaml
+$KUBECTL wait -n kubevirt-hyperconverged hco kubevirt-hyperconverged --for=condition=Available --timeout=5m
+
 hco_bucket="kubevirt-prow/devel/nightly/release/kubevirt/hyperconverged-cluster-operator"
 echo "${build_date}" > build-date
-echo "${IMAGE_REGISTRY}/${IMAGE_PREFIX}/hco-container-registry:${IMAGE_TAG}" > hco-bundle
+echo "${BUNDLE_IMAGE_NAME}" > hco-bundle
+echo "${INDEX_IMAGE_NAME}" > hco-index
```
Comment (orenc1) [inline automation/nightly/test-nightly-build.sh:117]: looks like you're writing to the files `hco-bundle` and `hco-index`, but in the following commands you're using them with the `-image` suffix.

File: automation/nightly/test-nightly-build.sh (hunk 1 above)
Comment (nunnatsa) [reply automation/nightly/test-nightly-build.sh:73]: good catch! fixing

File: automation/nightly/test-nightly-build.sh [hunk 3]
```diff
@@ -10,24 +10,34 @@ wget -q https://dl.google.com/go/go1.22.6.linux-amd64.tar.gz
 tar -C /usr/local -xf go*.tar.gz
 export PATH=/usr/local/go/bin:$PATH
 
-# get latest commits
+# get latest KubeVirt commits
```
Comment (orenc1) [inline automation/nightly/test-nightly-build.sh:13]: why `commits` in plural, if you get only the latest one (single)?
same for CDI below

**Review by orenc1 — APPROVED** [review]
/approve

**Comment by nunnatsa** [comment]
hco-e2e-kv-smoke-gcp lane passed
/override ci/prow/hco-e2e-kv-smoke-azure

**Pull request merged**